client/src/app/page/common/tasks.service.ts

```diff
@@ -1,16 +1,65 @@
 import { Injectable } from '@angular/core';
-import { Observable, of } from 'rxjs';
+import { Observable, of} from 'rxjs';
 import { Task } from './task';
-import { TASKS } from './mock-tasks';
+import { Http, RequestOptions, Headers } from '@angular/http';
+import { UserService, api } from '../../app_services/user.service';
+import * as moment from 'moment';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/catch';
+import { throwError } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
+
 export class TasksService {
 
-  constructor() { }
+  constructor(private http: Http, private userService: UserService) {}
 
   getTasks(): Observable<Task[]> {
-    return of(TASKS);
+    const options = this.userService.getRequestOptions();
+    return this.http.get(`${api}/issues/all`, options)
+    .map(response => {
+      const getTask = response.json();
+      const tasks: Task[] = getTask.map( (item: any) => {
+        return {
+          id: item._id,
+          name: item.title,
+          excerpt: '',
+          status: {name: item.priority, value: this.getStatusValue(item.priority)},
+          type: {name: item.type, value: this.getTaskType(item.type)},
+          date: this.convertDate(item.created),
+          author: '',
+          content: item.value
+        };
+      });
+
+      return tasks;
+    });
+  }
+
+  getStatusValue = (status: string): number => {
```

Comment:
use enum for this.

enum Status = {
 .... 
}

....

if (status  === Status.Normal) ...